OpenRA.Game/Actor.cs

```diff
@@ -454,6 +511,70 @@ public IEnumerable<WPos> GetTargetablePositions()
 			return new[] { CenterPosition };
 		}
 
+		#region Conditions
+
+		void UpdateConditionState(string condition, int token, bool isRevoke)
+		{
+			ConditionState conditionState;
+			if (!conditionStates.TryGetValue(condition, out conditionState))
+				return;
+
+			if (isRevoke)
+				conditionState.Tokens.Remove(token);
+			else
+				conditionState.Tokens.Add(token);
+
+			conditionCache[condition] = conditionState.Tokens.Count;
+
+			// Wait until all traits are created.
+			if (created)
+				foreach (var notify in conditionState.Notifiers)
+					notify(this, readOnlyConditionCache);
+		}
+
+		/// <summary>Grants a specified condition.</summary>
+		/// <returns>The token that is used to revoke this condition.</returns>
+		public int GrantCondition(string condition)
+		{
+			var token = nextConditionToken++;
+			conditionTokens.Add(token, condition);
+
+			// Conditions may be granted before the state is initialized.
+			// These conditions will be processed in INotifyCreated.Created.
+			if (conditionStates != null)
+				UpdateConditionState(condition, token, false);
+
+			return token;
+		}
+
+		/// <summary>
+		/// Revokes a previously granted condition.
+		/// </summary>
+		/// <param name="token">The token ID returned by GrantCondition.</param>
+		/// <returns>The invalid token ID.</returns>
+		public int RevokeCondition(int token)
+		{
+			string condition;
+			if (!conditionTokens.TryGetValue(token, out condition))
+				throw new InvalidOperationException("Attempting to revoke condition with invalid token {0} for {1}.".F(token, this));
+
+			conditionTokens.Remove(token);
+
+			// Conditions may be granted and revoked before the state is initialized.
+			if (conditionStates != null)
```

Comment:
Same here.

Reply:
It's left-overs from copy-pasting - same as above.